guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/GuvnorM2Repository.java

```diff
@@ -509,11 +513,9 @@ private RemoteRepository getRemoteRepoFromDeployment( DeploymentRepository repo
         Server server = settings.getServer( repo.getId() );
 
         if ( server != null ) {
-            remoteRepoBuilder.setAuthentication(
-                    new AuthenticationBuilder()
-                            .addUsername( server.getUsername() )
-                            .addPassword( server.getPassword() )
-                            .build() );
+            Authentication authentication = embedder.getMavenSession().getRepositorySession()
+                    .getAuthenticationSelector().getAuthentication( remoteRepoBuilder.build() );
```

Comment:
Hmm, the aether javadoc is the orginal from 2011.  May have been that way for Eclipse, but Maven uses AuthenticationSelector is same way while building the project (regardless of mojos executed) at [AuthenticationSelector (line 246)](https://git-wip-us.apache.org/repos/asf?p=maven.git;a=blob;f=maven-core/src/main/java/org/apache/maven/bridge/MavenRepositorySystem.java;h=93317105eb196040fc310a28a0e4c00c2bd36ca8;hb=HEAD).  This was part of inspiration to use AuthenticationSelector, as the Authentication itself is encoded in memory, and reading the settings-security.xml to manually decrypt passwords would duplicate work already accomplished by DefaultMaven (invoked in MavenEmbedder constructor).